### PR TITLE
Updating project templates for isolated worker model to use latest recommendations

### LINKED ISSUE #1483: Update project template for isolated worker model to use latest recommendations
e.g., Framework reference to ASP.NET Core (unless .NET Framework)

### PR DESCRIPTION
**Pull request by surgupta-msft**
resolves #1483 

- Added Framework reference to ASP.NET Core (unless .NET Framework)
- Verified and updated nuget packages to the latest versions
- Updated both csharp and fsharp templates

### CODE REVIEW

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj [hunk 1]
```diff
@@ -1,17 +1,18 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (mattchenderson) [inline Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:3]: Shouldn't this also just be `TargetFrameworkValue`? I'm not sure why this was `net6.0` before...

File: Functions.Templates/ProjectTemplate_v4.x/CSharp-Isolated/Company.FunctionApp.csproj [hunk 2]
```diff
@@ -10,10 +10,13 @@
     <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.FunctionApp</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
+  <!--#if (NetCore)-->
+  	<FrameworkReference Include="Microsoft.AspNetCore.App" />
+  <!--#endif -->
     <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.20.0" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.2" />
```
Comment (surgupta-msft) [inline Functions.Templates/ProjectTemplate_v4.x/CSharp-Isolated/Company.FunctionApp.csproj:17]: Looks like the latest version for this package is 1.16.3 released 4 days back - https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Sdk. I will update this.

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
     <AzureFunctionsVersion>AzureFunctionsVersionValue</AzureFunctionsVersion>
     <OutputType>Exe</OutputType>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
     <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.FunctionApp</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Comment (mattchenderson) [inline Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:11]: This should have the same conditional

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
     <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.20.0" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.2" />
```
Comment (surgupta-msft) [inline Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:13]: Update needed here to 1.16.3

File: Functions.Templates/ProjectTemplate_v4.x/CSharp-Isolated/Company.FunctionApp.csproj (hunk 2 above)
Comment (mattchenderson) [inline Functions.Templates/ProjectTemplate_v4.x/CSharp-Isolated/Company.FunctionApp.csproj:17]:
```suggestion
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.3" />
```

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj (hunk 4 above)
Comment (mattchenderson) [inline Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:13]:
```suggestion
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.3" />
```

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj (hunk 1 above)
Comment (surgupta-msft) [reply Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:3]: Yeah I am not sure. Tagging @liliankasem, @kshyju

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj (hunk 3 above)
Comment (surgupta-msft) [reply Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj]: I will update this if we decide to change TargetFramework to `TargetFrameworkValue`

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj (hunk 1 above)
Comment (liliankasem) [reply Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:3]: I haven't touched the proj version - @soninaren ?

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <AzureFunctionsVersion>AzureFunctionsVersionValue</AzureFunctionsVersion>
     <OutputType>Exe</OutputType>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
     <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.FunctionApp</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
+  	<FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Comment (liliankasem) [inline Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:11]: nit: whitespace not in alignment with rest of the references

File: Functions.Templates/ProjectTemplate_v4.x/CSharp-Isolated/Company.FunctionApp.csproj [hunk 6]
```diff
@@ -10,10 +10,13 @@
     <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.FunctionApp</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
+  <!--#if (NetCore)-->
+  	<FrameworkReference Include="Microsoft.AspNetCore.App" />
+  <!--#endif -->
     <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.20.0" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.3" />
```
Comment (kshyju) [inline Functions.Templates/ProjectTemplate_v4.x/CSharp-Isolated/Company.FunctionApp.csproj:17]: https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Sdk/1.16.4 is out now. Let's use that.

File: Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj (hunk 1 above)
Comment (mattchenderson) [reply Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj:3]: Traces back to https://github.com/Azure/azure-functions-templates/pull/1074.

The initial state was arguably wrong, and it was bumped once. I still assert that it should use the templating engine parameter for this.

Comment (surgupta-msft) [reply Functions.Templates/ProjectTemplate_v4.x/FSharp-Isolated/Company.FunctionApp.fsproj]: Got it. Updated to use the parameter.

**Review by liliankasem — APPROVED** [review]

**Pull request merged**